96.md

```diff
@@ -0,0 +1,192 @@
+NIP-96
+======
+
+HTTP File Storage Integration
+-----------------------------
+
+`draft` `optional` `author:arthurfranca` `author:Semisol` `author:staab`
+
+## Introduction
+
+This NIP defines a REST API for HTTP file storage servers intended to be used in conjunction with the nostr network.
+The API will enable nostr users to upload files and later reference them by url on nostr notes.
+
+The spec DOES NOT use regular nostr events through websockets for
+storing, requesting nor retrieving data because, for simplicity, the server
+will not have to learn anything about nostr relays.
+
+## Server Adaptation
+
+File storage servers wishing to be accessible by nostr users should opt-in by making available
+an https route at `/.well-known/nostr.json` with `nip96.routes` field in the following form:
+
+```js
+{
+  "nip96": {
+    // required
+    "routes": {
+      "upload": "https://upload.domain/upload-path",
+      "download": "https://download.domain/download-path",
+      "deletion": "https://deletion.domain/deletion-path"
+    },
```

Comment:
I though this too at first but then at the time I thought it would be easier to understand with 3 separate keys.

So you want A) or B)?

A)
```
  "routes": {
      "api": "https://api.domain/upload-and-deletion-path",
      "download": "https://download.domain/download-path"
   }
```

B)
```
  "routes": {
    "upload/deletion": "https://upload.and.deletion.domain/upload-and-deletion-path",
    "download": "https://download.domain/download-path"
  }
```

Reply:
Using "api_url" (that may include any custom path) and "download_url"